plugin-server/src/main/pluginsServer.ts

```diff
@@ -428,6 +433,27 @@ export async function startPluginsServer(
             }
         }
 
+        if (capabilities.personOverrides) {
+            const postgres = hub?.postgres ?? new PostgresRouter(serverConfig)
+            const kafkaProducer = hub?.kafkaProducer ?? (await createKafkaProducerWrapper(serverConfig))
+            const overridesWriter = new DeferredPersonOverrideWriter(postgres)
```

Comment:
It's not needed for the other part of this process (`addPersonOverrides` in the ingestion consumer) so it didn't seem like it made sense to add it as a required parameter to the constructor, and making it optional didn't seem to make much sense for the opposite reason as its required here. Here's the other code:

https://github.com/PostHog/posthog/blob/7e2065a43cb218768cf522e3d9b659397d52ac37/plugin-server/src/worker/ingestion/person-state.ts#L717-L743

I think there's a reasonable argument that maybe they shouldn't be part of the same class, but also a reasonable argument for keeping them close together since they're inextricably related. No specific approach stood out to me as particularly better than any other one so I just took the one that seemed least likely to result in the `add` and `process` steps getting out of sync over time if modified.

Reply:
Split up the writer and worker with 3bc229092523246a8ed34ae7e5fc2d2a115029db and made the producer a constructor parameter with 7c1ad872405af0b9b9ed94b81f12ef847be0e9a8 instead. That seems like a better overall approach all things considered, thanks for pointing out the strangeness of this call.